pallets/accounts/src/lib.rs

```diff
@@ -407,18 +474,70 @@ pub mod pallet {
 			ensure!(!<EVMLinks<T>>::contains_key(&sender), Error::<T>::AccountAlreadyLinked);
 			ensure!(!<EVMLinksReverse<T>>::contains_key(eth_key), Error::<T>::AccountAlreadyLinked);
 
+			if <EthReservedTickets<T>>::contains_key(&eth_key) {
+				let tickets_amount = <EthReservedTickets<T>>::get(&eth_key);
+				if let Some(amount) = tickets_amount {
+					ensure!(!amount.is_zero(), Error::<T>::SystematicFailure);
+				}
+			}
+			if <EthReservedNova<T>>::contains_key(&eth_key) {
+				let nova_amount = <EthReservedNova<T>>::get(&eth_key);
+				if let Some(amount) = nova_amount {
+					ensure!(!amount.is_zero(), Error::<T>::SystematicFailure);
+					let nova_old_balance =
+						pallet_balances::Pallet::<T>::free_balance(&sender.clone());
+					ensure!(
+						nova_old_balance + amount >=
+							<pallet_balances::Pallet<T> as Currency<T::AccountId>>::minimum_balance(),
+						Error::<T>::BelowMinimumBalance
+					);
+					ensure!(
+						!nova_old_balance.checked_add(&amount).is_none(),
+						Error::<T>::SystematicFailure
+					);
+				}
+			}
+
+			// Check if the Ethereum account has already some tickets and nova registered when it was not linked
+			if <EthReservedTickets<T>>::contains_key(&eth_key) {
```

Comment:
Can these duplicate if branches be merged?
It doesn't look like they have any specific ordering dependency

Reply:
There is no ordering needed in this case. All reserved tokens will be handled the same.

Reply:
I'd just merge the if statements then, currently it reads:
```
if <EthReservedTickets<T>>::contains_key(&eth_key) { ... }
if <EthReservedNova<T>>::contains_key(&eth_key) { ...}
if <EthReservedTickets<T>>::contains_key(&eth_key) { ... }
if <EthReservedNova<T>>::contains_key(&eth_key) { ...}
```
change it to
```
if <EthReservedTickets<T>>::contains_key(&eth_key) { ... }
if <EthReservedNova<T>>::contains_key(&eth_key) { ...}
```

Reply:
So, actually I made this on purpose and it is related to how storage in Substrate works. They were merged before a previous review I received (so hopefully this is fine now). In Substrate once the state/storage is written it would not rollback in case the transction fails later. So here the first two if-statements check that there would be no problem to perform the `mint_into()` function calls in the third and fourth if-statements. If they were merged, and the second if-statement fails after the first if-statement was successful, the storage written in the first if-statement would not rollback, causing state incoherence.

Reply:
Yeah then it makes sense to have it this way